### ISSUE TITLE
the menu should support hosts with a custom port

### ISSUE THREAD

**Issue by SkySkimmer** [RequestPolicyContinued/requestpolicy#649]
See http://a.pomf.se/xjbnjr.png (requests from https://forums.sufficientvelocity.com/threads/arctic-freeze-cycle-maximum-reached-february-15th.15383/page-4 to http://www.iup.uni-bremen.de:8084/amsr2data/asi_daygrid_swath/n6250/2014/may/asi-AMSR2-n6250-20140515-v5_nic.png)

**Comment by nodiscc** [comment]
Hi @SkySkimmer I can't reproduce your problem right now on https://forums.sufficientvelocity.com/threads/arctic-freeze-cycle-maximum-reached-february-15th.15383/page-4.

Do you have another rule that matches `uni-bremen.de` as a destination in your current rules?

**Comment by SkySkimmer** [comment]
Yes. When I remove it through the manage policies page I get the origin to destination option back in the menu, then when I add it again the bug comes back.

**Comment by myrdd** [comment]
Confirmed.

*Default Policy:* Deny.
*Rule:* allow `*.sufficientvelocity.com` to `*.uni-bremen.de`

There's something I first thought which would be another bug, but it isn't. The requests are blocked by the default policy, which means that *the rule didn't match*. The reason is the custom port of the destination. A rule that does not specify a port matches only the standard port.

The actual issue is that the menu can't handle hosts with a custom port. Some logic thinks that there *is* a matching rule, other logic thinks there *isn't*.

---------------

When there is an "allow" rule from `*.sufficientvelocity.com` to `*.uni-bremen.de:8084` (port specified), then the menu looks as follows:

![menu](https://cloud.githubusercontent.com/assets/4286829/7693631/9452cacc-fdd4-11e4-9f96-7f496035701a.jpg)

**Comment by myrdd** [comment]
Closing in favor of #473
